### DISCUSSION TITLE
New Option for Interface

### DISCUSSION THREAD

**Discussion by mleo2003** [imartinez/privateGPT#484] [Ideas]
Most people have realized that, without semi decent hardware to run this with special setup for GPU support, this project can take some time. While others see this as a problem, I wonder if we can't turn this into almost a "feature".

How I imagine that working, is if instead of treating this like ChatGPT, we did it more like "EmailGPT". Set up the script to watch an email mailbox, use the from/subject as a way to identify context if needed, and the body as the prompt. Email is already pretty "asynchronous", so waiting for a response to a question feels right.

I can also see this being a way to make queries to private data "remotely", though that also hinges on how secure the transport is (i.e. not very). But if done internally (say, have the script use a gmail inbox and send it emails from another gmail), it could be better than other options.

Just a random thought I had about a way to make the waiting better, use it as an email answering service. I'd love for things to speed up even more, but this could help some do this even from a mobile device.

**Comment by hcchengithub** [comment]
Good idea, but why not simply use ChatGPT-4 as the AI for privateGPT?
I don't care exposing sensitive data to OpenAI1. How to modify .env to choose ChatGPT-4 as the AI for privateGPT?
